### PR TITLE
Improve release process description

### LINKED ISSUE #39: R release section
Some questions/issues regarding the release section in:
https://github.com/Open-Systems-Pharmacology/developer-docs/blob/main/ospsuite-r-specifics/collaboration_guide.md#releasing-versions

grafik

1. "the release is done" - the release still needs to be created on GitHub, right?
2. replace "should" with "must"
3. Don't understand "Repeat steps 2 to 6". When the release is done, there is nothing to do. Would remove this bullet point completely

### PR DESCRIPTION
**Pull request by Felixmil**
closes #39

### CODE REVIEW

File: ospsuite-r-specifics/collaboration_guide.md [hunk 1]
```diff
@@ -86,36 +88,73 @@ This workflow implies that:
...
+  What should the new version be? (0 to exit) 
+  
+  1: major --> X+1.Y.Z
+  2: minor --> X.Y+1.Z
+  3: patch --> X.Y.Z+1
+  4:   dev --> X.Y.Z.9000
+  ```
 
-8. Repeat steps 2 to 6.
+9. Repeat steps 2 to 6.
```
Comment (Yuri05) [inline ospsuite-r-specifics/collaboration_guide.md:158]: Still don't understand why we need this "Repeat steps". Once the release is done - we are done.

Comment (Felixmil) [reply ospsuite-r-specifics/collaboration_guide.md:158]: No, to finish the release process, we need to put the main branch back to development mode which means:
- create a dedicated branch
- add the `.9000` version identifier (DESCRIPTION and NEWS)
- commit and push
And these steps are covered in steps 2 to 5

**Review by rengelke — COMMENTED** [review]
Can we cover how to upload or handle binaries as part of the release.

**Comment by Felixmil** [comment]
> Can we cover how to upload or handle binaries as part of the release.

@rengelke done in f5d67dd before we implement an automated way to do so.

**Pull request merged**